src/App.tsx

```diff
@@ -1,54 +1,95 @@
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropDownActive, setIsDropDownActive] = useState(false);
+  const delay = 1000;
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, delay),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setQuery(value);
+    applyQuery(value);
+  };
+
+  const handleClick = (person: Person) => {
+    setSelectedPerson(person);
+    setQuery(person.name);
+    setIsDropDownActive(false);
+  };
+
+  const filteredPeople = useMemo(() => {
+    if (!appliedQuery) {
+      setSelectedPerson(null);
+      setIsDropDownActive(false);
+    } else {
+      setIsDropDownActive(true);
+    }
+
+    return peopleFromServer.filter((
+      { name },
+    ) => name.toLowerCase().includes(appliedQuery.toLowerCase()));
+  }, [peopleFromServer, appliedQuery]);
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson
+          ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+          : 'No selected person'}
       </h1>
 
-      <div className="dropdown is-active">
+      <div
+        className={classNames('dropdown', {
+          'is-active': isDropDownActive,
+        })}
+      >
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
             className="input"
+            value={query}
+            onChange={handleQueryChange}
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
           <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+            {filteredPeople.length > 0 ? (
+              filteredPeople.map(person => (
+                <div
+                  className="dropdown-item"
+                  key={person.slug}
+                >
+                  <tr
+                    className={classNames({
```

Comment:
You are using a 'tr' element outside of a 'table'. This is incorrect in HTML semantics. Consider replacing it with another valid HTML element such as a 'div' or 'p' element.